### ISSUE TITLE
NullPointerException in Scratch window in PhpStorm EAP 9 #PS-141.1000

### ISSUE THREAD

**Issue by ragol** [psliwa/idea-composer-plugin#7]
Today I have upgraded Composer Plugin from v1.0.4  to v1.0.5. Since the upgrade I get an `NullPointerException` every couple of minutes while editing JavaScript in a scratch window. Most (all?) of the time it happens when I select the surrounding context by hitting CTRL-W.

The error message is just "null". Details below:

```
null
java.lang.NullPointerException
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.loop$2(SchemaDocumentationProvider.scala:69)
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.findTokens(SchemaDocumentationProvider.scala:73)
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.findProperty(SchemaDocumentationProvider.scala:37)
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.generateDoc(SchemaDocumentationProvider.scala:31)
	at com.intellij.lang.documentation.CompositeDocumentationProvider.generateDoc(CompositeDocumentationProvider.java:134)
	at com.intellij.xml.util.documentation.HtmlDocumentationProvider.generateDoc(HtmlDocumentationProvider.java:161)
	at com.intellij.lang.documentation.CompositeDocumentationProvider.generateDoc(CompositeDocumentationProvider.java:134)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$9$1.run(CtrlMouseHandler.java:630)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:872)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$9.run(CtrlMouseHandler.java:626)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:298)
	at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:295)
	at com.intellij.util.concurrency.QueueProcessor$2$1.run(QueueProcessor.java:110)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:107)
	at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:104)
	at com.intellij.util.concurrency.QueueProcessor$3$1.run(QueueProcessor.java:215)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor$3.run(QueueProcessor.java:212)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)



java.lang.NullPointerException
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.loop$2(SchemaDocumentationProvider.scala:69)
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.findTokens(SchemaDocumentationProvider.scala:73)
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.findProperty(SchemaDocumentationProvider.scala:37)
	at org.psliwa.idea.composerJson.intellij.codeAssist.schema.SchemaDocumentationProvider.generateDoc(SchemaDocumentationProvider.scala:31)
	at com.intellij.lang.documentation.CompositeDocumentationProvider.generateDoc(CompositeDocumentationProvider.java:134)
	at com.intellij.xml.util.documentation.HtmlDocumentationProvider.generateDoc(HtmlDocumentationProvider.java:161)
	at com.intellij.lang.documentation.CompositeDocumentationProvider.generateDoc(CompositeDocumentationProvider.java:134)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$9$1.run(CtrlMouseHandler.java:630)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:872)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$9.run(CtrlMouseHandler.java:626)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:298)
	at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:295)
	at com.intellij.util.concurrency.QueueProcessor$2$1.run(QueueProcessor.java:110)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:107)
	at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:104)
	at com.intellij.util.concurrency.QueueProcessor$3$1.run(QueueProcessor.java:215)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor$3.run(QueueProcessor.java:212)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)
```

**Comment by psliwa** [comment]
Thanks! This happens only when you are editing scratch javascript file?

**Comment by ragol** [comment]
I hope so. At least I have not seen it in other windows / files. On the other hand, the exception gets thrown sporadically even in a scratch javascript file. It might be thrown multiple times in a minute or, as it is at the moment, do not show up for half an hour.

**Comment by psliwa** [comment]
I have fixed this and two more similar bugs in `1.0.6` release.
